implementations/rust/kex/src/x3dh.rs

```diff
@@ -274,7 +279,7 @@ impl KeyExchanger for X3dhResponder {
                 let ek = PublicKey::Curve25519(*array_ref![data, 0, 32]);
 
                 let dh1 = vault.ec_diffie_hellman(self.signed_prekey, *eik)?;
-                let dh2 = vault.ec_diffie_hellman(self.identity_key, ek)?;
+                let dh2 = vault.ec_diffie_hellman(self.identity_key.unwrap(), ek)?;
```

Comment:
does this need to be handled rather than unwrapping?

Reply:
At this point in the code it's always `Some`

Reply:
At this point, _maybe_ given the current logic -- but what if that changes? Could you point me to what guarantees this as `Some`? Sorry if I'm missing it. If it's just match logic, I don't think that's enough.

Reply:
See line 141

Reply:
ok, so you should definitely use the value if you can, NOT continuing to call `.unwrap()` everywhere. assigning the result of the `vault.secret_generate(p_atts)?` call to a new variable will ensure that at no point in the future does that `self.identity_key` happen to turn into a `None`... code changes & this is only valid in it's current state.